src/main/java/athleticli/data/sleep/SleepGoal.java

```diff
@@ -1,8 +1,87 @@
-/**
- * To be implemented in future version of AthletiCLI.
- */
-
 package athleticli.data.sleep;
 
-public class SleepGoal {
+import athleticli.data.Data;
+import athleticli.data.Goal;
+
+import java.time.format.DateTimeFormatter;
+import java.util.Locale;
+
+
+
+public class SleepGoal extends Goal {
+
+    public static final DateTimeFormatter TIME_FORMATTER = DateTimeFormatter.ofPattern("HH:mm",
+        Locale.ENGLISH);
+
+    public enum GoalType {
+        DURATION, STARTTIME, ENDTIME
+    }
+
+    private final GoalType goalType;
+    private int target;
+
+    /**
+     * Constructs a sleep goal.
+     * @param timespan    The timespan of the sleep goal.
+     * @param goalType    The goal type of the sleep goal.
+     * @param targetValue The target value of the sleep goal in minutes. (Used if goalType is DURATION)
+     * @param targetTime  The target time of the sleep goal. (Used if goalType is STARTTIME or ENDTIME)
+     */
+    public SleepGoal(GoalType goalType, TimeSpan timespan, int target) {
+        super(timespan);
+        this.target = target;
+        this.goalType = goalType;
+    }
```

Comment:
```suggestion
    /**
     * Constructs a sleep goal.
     * @param timeSpan    The time span of the sleep goal.
     * @param goalType    The goal type of the sleep goal.
     * @param targetValue The target value of the sleep goal in minutes. (Used if goalType is DURATION)
     * @param targetTime  The target time of the sleep goal. (Used if goalType is STARTTIME or ENDTIME)
     */
    public SleepGoal(GoalType goalType, TimeSpan timeSpan, int target) {
        super(timeSpan);
        this.target = target;
        this.goalType = goalType;
    }
```
Write `timeSpan` instead of `timespan` to be consistent with #120.

Reply:
Thanks for the good spot!